config/routes.rb

```diff
@@ -6,6 +6,10 @@
   root to: 'users#index'
   resources :users, only: %i[edit update show index create]
   resources :projects, only: %i[show]
+  resources :users, only: [] do
+    get '/calendar.ics' => 'users#ics_export', :as => 'ics_export'
+  end
```

Comment:
Seems like you can use  `get 'users/calendar.ics' => 'users#ics_export', :as => 'ics_export'` in oneline

Reply:
users/:user_id/calendar.ics